iamy/aws.go

```diff
@@ -129,7 +129,21 @@ func (a *AwsFetcher) fetchIamData() error {
 	if err != nil {
 		return err
 	}
-
+	// Fetch instance profiles
+	err = a.iam.ListInstanceProfilesPages(&iam.ListInstanceProfilesInput{},
+		func(resp *iam.ListInstanceProfilesOutput, lastPage bool) bool {
+			populateInstanceProfileErr = a.populateInstanceProfileData(resp)
+			if populateInstanceProfileErr != nil {
+				return false
+			}
+			return true
+		})
+	if populateInstanceProfileErr != nil {
```

Comment:
does this need to be here? looks like you've already got a check for this on L136.

Reply:
I think that's right, the L136 is to escape the iterator function

Reply:
Yep, it needs to be there.
The function that includes line 136 is called by the paging function a.iam.ListInstanceProfilesPages with each back results.   So the return false is effectively the break.